### PR TITLE
[Magiclys] Forge Demon Patrol Cycle

### PR DESCRIPTION
**Pull request by Maleclypse**
#### Summary
Mods "Make forge demons patrol the forge"

#### Purpose of change
Add some life to the forge demons lifecycle.

#### Describe the solution
uses spawn data `"patrol"'   I'm not having much luck with that as I want them mark the edges of the vault.

#### Describe alternatives you've considered
Leaving them as is.

#### Testing
testing is not going well.  They mostly cluster in a corner determined by the facing of the building rotation.

### CODE REVIEW

**Comment by Maleclypse** [comment]
@Venera3 Is there a way for me to make this work or do I need to make the corners they are trying to reach the exact corners of the vault?  And if so how do I determine that given that they spawn randomly across the spread of the vault space?

File: data/mods/Magiclysm/worldgen/forge_of_wonders.json [hunk 1]
```diff
@@ -85,8 +85,16 @@
...
-        { "monster": "GROUP_FORGEDEMONS", "x": [ 80, 90 ], "y": [ 8, 14 ], "density": 0.1, "repeat": [ 3, 6 ] }
+        { "monster": "GROUP_FORGE_DWARVES", "x": [ 24, 47 ], "y": [ 40, 45 ], "density": 0.1, "repeat": [ 3, 6 ] }
+      ],
+      "place_monster": [
+        {
+          "monster": "mon_forgedemons",
+          "x": [ 81, 90 ],
+          "y": [ 8, 14 ],
+          "repeat": [ 10, 18 ],
+          "spawn_data": { "patrol": [ { "x": 10, "y": 10 }, { "x": -10, "y": -10 }, { "x": 10, "y":-10 }, { "x": -10, "y": 10 } ] }
```
Comment (Maleclypse) [inline data/mods/Magiclysm/worldgen/forge_of_wonders.json:96]:
```suggestion
          "spawn_data": { "patrol": [ { "x": -3, "y": -1 }, { "x": -3, "y": 2 }, { "x": 3, "y": 2 }, { "x": 3, "y": -2 } ] }
```

**Comment by Maleclypse** [comment]
At this point I'm wanting them to go back to wherever they spawned at anytime they can't see someone to be hostile to.   But they always cluster up on a wall whenever they don't see me.

**Comment by Venera3** [comment]
I think to return them to their original position you'd need to hand-place them one by one (and assign a single-point "patrol route" to the same position).

**Comment by Maleclypse** [comment]
> I think to return them to their original position you'd need to hand-place them one by one (and assign a single-point "patrol route" to the same position).

So I've created some that should return to their point of origin and some that should wander normally.   The ones wandering normally work well.  The patrolling ones still try to run off into the distance.   Am I misunderstanding how their patrol point works?   Should I set their patrol point as 0,0 to get them to return to their spawn point?

**Pull request closed without merging**